hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/slowlog/TestOnlineSlowLogProvider.java

```diff
@@ -0,0 +1,307 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(TestOnlineSlowLogProvider.class);
+
+  private static final HBaseTestingUtility HBASE_TESTING_UTILITY = new HBaseTestingUtility();
+
+  private OnlineSlowLogProvider onlineSlowLogProvider;
+
+  private Configuration getOnlineSlowLogConf(int eventSize) {
+    Configuration conf = HBASE_TESTING_UTILITY.getConfiguration();
+    conf.setBoolean(HConstants.SLOW_LOG_BUFFER_ENABLED_KEY, true);
+    conf.setInt(HConstants.SLOW_LOG_RING_BUFFER_SIZE, eventSize);
+    return conf;
+  }
+
+  private SlowLogPayload getSlowLogPayload(int i) {
+    SlowLogPayload slowLogPayload = SlowLogPayload.newBuilder()
+      .setCallDetails("call_" + i)
+      .setClientAddress("client_" + i)
+      .setMethodName("method_" + i)
+      .setUserName("userName_" + i)
+      .setMultiGets(i)
+      .setMultiMutations(i)
+      .setMultiServiceCalls(i)
+      .setProcessingTime(i + 500)
+      .setQueueTime(i + 400)
+      .setResponseSize(i + 200)
+      .setStartTime(EnvironmentEdgeManager.currentTime())
+      .setServerClass("class_" + i)
+      .setParam("param_" + i)
+      .build();
+    return slowLogPayload;
+  }
+
+  /**
+   * confirm that for a ringbuffer of slow logs, payload on given index of buffer
+   * has expected elements
+   *
+   * @param i               index of ringbuffer logs
+   * @param j               data value that was put on index i
+   * @param slowLogPayloads list of payload retrieved from {@link OnlineSlowLogProvider}
+   */
+  private void confirmPayloadParams(int i, int j, List<SlowLogPayload> slowLogPayloads) {
+    Assert.assertEquals(slowLogPayloads.get(i).getClientAddress(), "client_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getCallDetails(), "call_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getMethodName(), "method_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getUserName(), "userName_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getProcessingTime(), j + 500);
+    Assert.assertEquals(slowLogPayloads.get(i).getQueueTime(), j + 400);
+    Assert.assertEquals(slowLogPayloads.get(i).getResponseSize(), j + 200);
+    Assert.assertEquals(slowLogPayloads.get(i).getServerClass(), "class_" + j);
+    Assert.assertEquals(slowLogPayloads.get(i).getParam(), "param_" + j);
+  }
+
+  @Test
+  public void testOnlieSlowLogConsumption() throws Exception {
+    Configuration conf = getOnlineSlowLogConf(8);
+    onlineSlowLogProvider = new OnlineSlowLogProvider(conf);
+    Assert.assertEquals(onlineSlowLogProvider.getSlowLogPayloads().size(), 0);
+    LOG.debug("Initially ringbuffer of Slow Log records is empty");
+
+    int i = 0;
+
+    // add 5 records initially
+    for (; i < 5; i++) {
+      SlowLogPayload slowLogPayload = getSlowLogPayload(i + 1);
+      onlineSlowLogProvider.addSlowLogPayload(slowLogPayload);
+    }
+
+    Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
```

Comment:
Instead of a fixed sleep, how about using our `waitFor` test helper method, waiting on the condition of `slowLogPayloads.size() == 5`. This will make for a more robust test.